File: android/services-v73/src/main/java/com/liferay/mobile/android/v73/commercesubscriptionentry/CommercesubscriptionentryService.java
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.mobile.android.v73.commercesubscriptionentry;

import com.liferay.mobile.android.service.BaseService;
import com.liferay.mobile.android.service.JSONObjectWrapper;
import com.liferay.mobile.android.service.Session;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * @author Bruno Farache
 */
public class CommercesubscriptionentryService extends BaseService {

	public CommercesubscriptionentryService(Session session) {
		super(session);
	}

	public void deleteCommerceSubscriptionEntry(long commerceSubscriptionEntryId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceSubscriptionEntryId", commerceSubscriptionEntryId);

			_command.put("/commerce.commercesubscriptionentry/delete-commerce-subscription-entry", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		session.invoke(_command);
	}

	public JSONObject fetchCommerceSubscriptionEntry(long commerceSubscriptionEntryId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceSubscriptionEntryId", commerceSubscriptionEntryId);

			_command.put("/commerce.commercesubscriptionentry/fetch-commerce-subscription-entry", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONArray getCommerceSubscriptionEntries(long companyId, long userId, int start, int end, JSONObjectWrapper orderByComparator) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("companyId", companyId);
			_params.put("end", end);
			_params.put("start", start);
			_params.put("userId", userId);
			mangleWrapper(_params, "orderByComparator", "com.liferay.portal.kernel.util.OrderByComparator<com.liferay.commerce.model.CommerceSubscriptionEntry>", orderByComparator);

			_command.put("/commerce.commercesubscriptionentry/get-commerce-subscription-entries", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public JSONArray getCommerceSubscriptionEntries(long companyId, long groupId, long userId, int start, int end, JSONObjectWrapper orderByComparator) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("companyId", companyId);
			_params.put("end", end);
			_params.put("groupId", groupId);
			_params.put("start", start);
			_params.put("userId", userId);
			mangleWrapper(_params, "orderByComparator", "com.liferay.portal.kernel.util.OrderByComparator<com.liferay.commerce.model.CommerceSubscriptionEntry>", orderByComparator);

			_command.put("/commerce.commercesubscriptionentry/get-commerce-subscription-entries", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public Integer getCommerceSubscriptionEntriesCount(long companyId, long userId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("companyId", companyId);
			_params.put("userId", userId);

			_command.put("/commerce.commercesubscriptionentry/get-commerce-subscription-entries-count", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getInt(0);
	}

	public Integer getCommerceSubscriptionEntriesCount(long companyId, long groupId, long userId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("companyId", companyId);
			_params.put("groupId", groupId);
			_params.put("userId", userId);

			_command.put("/commerce.commercesubscriptionentry/get-commerce-subscription-entries-count", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getInt(0);
	}

	public JSONObject searchCommerceSubscriptionEntries(long companyId, JSONArray groupIds, long maxSubscriptionCycles, int subscriptionStatus, String keywords, int start, int end, JSONObjectWrapper sort) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("companyId", companyId);
			_params.put("end", end);
			_params.put("groupIds", checkNull(groupIds));
			_params.put("keywords", checkNull(keywords));
			_params.put("maxSubscriptionCycles", maxSubscriptionCycles);
			_params.put("start", start);
			_params.put("subscriptionStatus", subscriptionStatus);
			mangleWrapper(_params, "sort", "com.liferay.portal.kernel.search.Sort", sort);

			_command.put("/commerce.commercesubscriptionentry/search-commerce-subscription-entries", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject searchCommerceSubscriptionEntries(long companyId, long maxSubscriptionCycles, int subscriptionStatus, String keywords, int start, int end, JSONObjectWrapper sort) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("companyId", companyId);
			_params.put("end", end);
			_params.put("keywords", checkNull(keywords));
			_params.put("maxSubscriptionCycles", maxSubscriptionCycles);
			_params.put("start", start);
			_params.put("subscriptionStatus", subscriptionStatus);
			mangleWrapper(_params, "sort", "com.liferay.portal.kernel.search.Sort", sort);

			_command.put("/commerce.commercesubscriptionentry/search-commerce-subscription-entries", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject updateCommerceSubscriptionEntry(long commerceSubscriptionEntryId, int subscriptionLength, String subscriptionType, JSONObject subscriptionTypeSettingsUnicodeProperties, long maxSubscriptionCycles, int subscriptionStatus, int nextIterationDateMonth, int nextIterationDateDay, int nextIterationDateYear, int nextIterationDateHour, int nextIterationDateMinute, int deliverySubscriptionLength, String deliverySubscriptionType, JSONObject deliverySubscriptionTypeSettingsUnicodeProperties, long deliveryMaxSubscriptionCycles, int deliverySubscriptionStatus, int deliveryNextIterationDateMonth, int deliveryNextIterationDateDay, int deliveryNextIterationDateYear, int deliveryNextIterationDateHour, int deliveryNextIterationDateMinute) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceSubscriptionEntryId", commerceSubscriptionEntryId);
			_params.put("deliveryMaxSubscriptionCycles", deliveryMaxSubscriptionCycles);
			_params.put("deliveryNextIterationDateDay", deliveryNextIterationDateDay);
			_params.put("deliveryNextIterationDateHour", deliveryNextIterationDateHour);
			_params.put("deliveryNextIterationDateMinute", deliveryNextIterationDateMinute);
			_params.put("deliveryNextIterationDateMonth", deliveryNextIterationDateMonth);
			_params.put("deliveryNextIterationDateYear", deliveryNextIterationDateYear);
			_params.put("deliverySubscriptionLength", deliverySubscriptionLength);
			_params.put("deliverySubscriptionStatus", deliverySubscriptionStatus);
			_params.put("deliverySubscriptionType", checkNull(deliverySubscriptionType));
			_params.put("deliverySubscriptionTypeSettingsUnicodeProperties", checkNull(deliverySubscriptionTypeSettingsUnicodeProperties));
			_params.put("maxSubscriptionCycles", maxSubscriptionCycles);
			_params.put("nextIterationDateDay", nextIterationDateDay);
			_params.put("nextIterationDateHour", nextIterationDateHour);
			_params.put("nextIterationDateMinute", nextIterationDateMinute);
			_params.put("nextIterationDateMonth", nextIterationDateMonth);
			_params.put("nextIterationDateYear", nextIterationDateYear);
			_params.put("subscriptionLength", subscriptionLength);
			_params.put("subscriptionStatus", subscriptionStatus);
			_params.put("subscriptionType", checkNull(subscriptionType));
			_params.put("subscriptionTypeSettingsUnicodeProperties", checkNull(subscriptionTypeSettingsUnicodeProperties));

			_command.put("/commerce.commercesubscriptionentry/update-commerce-subscription-entry", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject updateSubscriptionStatus(long commerceSubscriptionEntryId, int subscriptionStatus) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceSubscriptionEntryId", commerceSubscriptionEntryId);
			_params.put("subscriptionStatus", subscriptionStatus);

			_command.put("/commerce.commercesubscriptionentry/update-subscription-status", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

}

File: windows/Liferay.SDK/Service/V73/Commercesubscriptionentry/CommercesubscriptionentryService.cs
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Liferay.SDK.Service.V73.Commercesubscriptionentry
{
	public class CommercesubscriptionentryService : ServiceBase
	{
		public CommercesubscriptionentryService(ISession session)
			: base(session)
		{
		}

		public async Task<dynamic> UpdateSubscriptionStatusAsync(long commerceSubscriptionEntryId, int subscriptionStatus)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceSubscriptionEntryId", commerceSubscriptionEntryId);
			_parameters.Add("subscriptionStatus", subscriptionStatus);

			var _command = new JsonObject()
			{
				{ "/commerce.commercesubscriptionentry/update-subscription-status", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task DeleteCommerceSubscriptionEntryAsync(long commerceSubscriptionEntryId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceSubscriptionEntryId", commerceSubscriptionEntryId);

			var _command = new JsonObject()
			{
				{ "/commerce.commercesubscriptionentry/delete-commerce-subscription-entry", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}

		public async Task<dynamic> FetchCommerceSubscriptionEntryAsync(long commerceSubscriptionEntryId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceSubscriptionEntryId", commerceSubscriptionEntryId);

			var _command = new JsonObject()
			{
				{ "/commerce.commercesubscriptionentry/fetch-commerce-subscription-entry", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetCommerceSubscriptionEntriesAsync(long companyId, long groupId, long userId, int start, int end, JsonObjectWrapper orderByComparator)
		{
			var _parameters = new JsonObject();

			_parameters.Add("companyId", companyId);
			_parameters.Add("groupId", groupId);
			_parameters.Add("userId", userId);
			_parameters.Add("start", start);
			_parameters.Add("end", end);
			this.MangleWrapper(_parameters, "orderByComparator", "$languageUtil.getJSONWrapperClassName($parameter.type)", orderByComparator);

			var _command = new JsonObject()
			{
				{ "/commerce.commercesubscriptionentry/get-commerce-subscription-entries", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetCommerceSubscriptionEntriesAsync(long companyId, long userId, int start, int end, JsonObjectWrapper orderByComparator)
		{
			var _parameters = new JsonObject();

			_parameters.Add("companyId", companyId);
			_parameters.Add("userId", userId);
			_parameters.Add("start", start);
			_parameters.Add("end", end);
			this.MangleWrapper(_parameters, "orderByComparator", "$languageUtil.getJSONWrapperClassName($parameter.type)", orderByComparator);

			var _command = new JsonObject()
			{
				{ "/commerce.commercesubscriptionentry/get-commerce-subscription-entries", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<long> GetCommerceSubscriptionEntriesCountAsync(long companyId, long groupId, long userId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("companyId", companyId);
			_parameters.Add("groupId", groupId);
			_parameters.Add("userId", userId);

			var _command = new JsonObject()
			{
				{ "/commerce.commercesubscriptionentry/get-commerce-subscription-entries-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<long> GetCommerceSubscriptionEntriesCountAsync(long companyId, long userId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("companyId", companyId);
			_parameters.Add("userId", userId);

			var _command = new JsonObject()
			{
				{ "/commerce.commercesubscriptionentry/get-commerce-subscription-entries-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<dynamic> SearchCommerceSubscriptionEntriesAsync(long companyId, IEnumerable<long> groupIds, long maxSubscriptionCycles, int subscriptionStatus, string keywords, int start, int end, JsonObjectWrapper sort)
		{
			var _parameters = new JsonObject();

			_parameters.Add("companyId", companyId);
			_parameters.Add("groupIds", groupIds);
			_parameters.Add("maxSubscriptionCycles", maxSubscriptionCycles);
			_parameters.Add("subscriptionStatus", subscriptionStatus);
			_parameters.Add("keywords", keywords);
			_parameters.Add("start", start);
			_parameters.Add("end", end);
			this.MangleWrapper(_parameters, "sort", "$languageUtil.getJSONWrapperClassName($parameter.type)", sort);

			var _command = new JsonObject()
			{
				{ "/commerce.commercesubscriptionentry/search-commerce-subscription-entries", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> SearchCommerceSubscriptionEntriesAsync(long companyId, long maxSubscriptionCycles, int subscriptionStatus, string keywords, int start, int end, JsonObjectWrapper sort)
		{
			var _parameters = new JsonObject();

			_parameters.Add("companyId", companyId);
			_parameters.Add("maxSubscriptionCycles", maxSubscriptionCycles);
			_parameters.Add("subscriptionStatus", subscriptionStatus);
			_parameters.Add("keywords", keywords);
			_parameters.Add("start", start);
			_parameters.Add("end", end);
			this.MangleWrapper(_parameters, "sort", "$languageUtil.getJSONWrapperClassName($parameter.type)", sort);

			var _command = new JsonObject()
			{
				{ "/commerce.commercesubscriptionentry/search-commerce-subscription-entries", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> UpdateCommerceSubscriptionEntryAsync(long commerceSubscriptionEntryId, int subscriptionLength, string subscriptionType, IDictionary<string, object> subscriptionTypeSettingsUnicodeProperties, long maxSubscriptionCycles, int subscriptionStatus, int nextIterationDateMonth, int nextIterationDateDay, int nextIterationDateYear, int nextIterationDateHour, int nextIterationDateMinute, int deliverySubscriptionLength, string deliverySubscriptionType, IDictionary<string, object> deliverySubscriptionTypeSettingsUnicodeProperties, long deliveryMaxSubscriptionCycles, int deliverySubscriptionStatus, int deliveryNextIterationDateMonth, int deliveryNextIterationDateDay, int deliveryNextIterationDateYear, int deliveryNextIterationDateHour, int deliveryNextIterationDateMinute)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceSubscriptionEntryId", commerceSubscriptionEntryId);
			_parameters.Add("subscriptionLength", subscriptionLength);
			_parameters.Add("subscriptionType", subscriptionType);
			_parameters.Add("subscriptionTypeSettingsUnicodeProperties", subscriptionTypeSettingsUnicodeProperties);
			_parameters.Add("maxSubscriptionCycles", maxSubscriptionCycles);
			_parameters.Add("subscriptionStatus", subscriptionStatus);
			_parameters.Add("nextIterationDateMonth", nextIterationDateMonth);
			_parameters.Add("nextIterationDateDay", nextIterationDateDay);
			_parameters.Add("nextIterationDateYear", nextIterationDateYear);
			_parameters.Add("nextIterationDateHour", nextIterationDateHour);
			_parameters.Add("nextIterationDateMinute", nextIterationDateMinute);
			_parameters.Add("deliverySubscriptionLength", deliverySubscriptionLength);
			_parameters.Add("deliverySubscriptionType", deliverySubscriptionType);
			_parameters.Add("deliverySubscriptionTypeSettingsUnicodeProperties", deliverySubscriptionTypeSettingsUnicodeProperties);
			_parameters.Add("deliveryMaxSubscriptionCycles", deliveryMaxSubscriptionCycles);
			_parameters.Add("deliverySubscriptionStatus", deliverySubscriptionStatus);
			_parameters.Add("deliveryNextIterationDateMonth", deliveryNextIterationDateMonth);
			_parameters.Add("deliveryNextIterationDateDay", deliveryNextIterationDateDay);
			_parameters.Add("deliveryNextIterationDateYear", deliveryNextIterationDateYear);
			_parameters.Add("deliveryNextIterationDateHour", deliveryNextIterationDateHour);
			_parameters.Add("deliveryNextIterationDateMinute", deliveryNextIterationDateMinute);

			var _command = new JsonObject()
			{
				{ "/commerce.commercesubscriptionentry/update-commerce-subscription-entry", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}
	}
}